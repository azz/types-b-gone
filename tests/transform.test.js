"use strict";

const fs = require("fs");
const relative = require("path").relative;
const glob = require("glob");

describe("types-b-gone", () => {
  let bGone;

  beforeEach(() => {
    bGone = require("../");
  });

  it("exports a function", () => {
    expect(typeof bGone).toBe("function");
  });

  it("given an empty string, returns a string", () => {
    expect(bGone("")).toEqual("");
  });

  glob.sync(__dirname + "/*/**/*.[jt]s").forEach(testFile => {
    const source = fs.readFileSync(testFile).toString();

    it(`transforms ${relative(__dirname, testFile)}`, () => {
      const output = bGone(source);
      const actual = raw(source + "~".repeat(80) + "\n" + output);
      expect(actual).toMatchSnapshot(testFile);
    });
  });
});

function raw(string) {
  if (typeof string !== "string") {
    throw new Error("Raw snapshots have to be strings.");
  }
  return { [Symbol.for("raw")]: string };
}
