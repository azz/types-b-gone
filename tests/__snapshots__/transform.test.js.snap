// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convert/enum.ts 1`] = `
enum Enum {
  A = 1,
  B,
  C
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
const Enum = __enum(["A", 1], "B", "C");
`;

exports[`sanity/javascript.js 1`] = `
"directive";

function f() {
  return y;
}

class X {
  constructor(...args) {
    super(...args);
  }

  method() {
    return y;
  }
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
"directive";

function f() {
    return y;
}

class X {
    constructor(...args) {
        super(...args);
    }

    method() {
        return y;
    }
}
`;

exports[`strip/functions.ts 1`] = `
function f<T>(a: number, b: (x) => y): void {}

declare function foo() {
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
function f(a, b) {}
`;

exports[`strip/generics.ts 1`] = `
class X extends Y<T> {}

class X<T> extends Y {}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
class X extends Y {}
class X extends Y {}
`;

exports[`strip/interface.ts 1`] = `
interface Foo {
  bar: string;
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

`;

exports[`strip/typeAnnotation.ts 1`] = `
var x: number;
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
var x;
`;
